app/src/main/java/io/intrepid/webservicesexercise/MainActivity.java

```diff
@@ -28,29 +38,44 @@ protected void onCreate(Bundle savedInstanceState) {
         ButterKnife.bind(this);
 
         Timber.plant(new Timber.DebugTree());
+
     }
 
+    private static GitHubApi createRestApi() {
+        OkHttpClient.Builder builder = new OkHttpClient().newBuilder();
+        OkHttpClient httpClient = builder
+                .connectTimeout(2, TimeUnit.SECONDS)
+                .build();
+        return new Retrofit.Builder()
+                .baseUrl(GITHUB_BASE_URL)
+                .client(httpClient)
+                .addConverterFactory(GsonConverterFactory.create())
+                .build()
+                .create(GitHubApi.class);
+    }
     @Override
     protected void onDestroy() {
         super.onDestroy();
-
-        if (getGitHubUserTask != null) {
-            getGitHubUserTask.setCallback(null);
-        }
     }
 
     @OnClick(R.id.submit_button)
     public void submitButtonClicked() {
-        getGitHubUserTask = new GetGitHubUserTask();
-        getGitHubUserTask.setCallback(this);
-        getGitHubUserTask.execute(usernameInputView.getText().toString());
-    }
+        Call<GitHubUser> call = createRestApi().getUser(usernameInputView.getText().toString());
+        call.enqueue(new Callback<GitHubUser>() {
+            @Override
+            public void onResponse(Call<GitHubUser> call, Response<GitHubUser> response) {
+                GitHubUser user = response.body();
+                Picasso.with(getApplicationContext())
```

Comment:
use context with nearest scope whenever possible -- i.e., use `MainActivity.this`